### PR TITLE
WIP Multiple Fixes and Enhancements

### PR DESCRIPTION
**Pull request by francistse23**
screenshots for those who dont want to pull the project:

preview1
preview2

Worked on what feels better from my perspective. If anything doesn't work out or there's a better way to show/do it, let me know and I'll fix it

## App
- Added max width to whole app (currently at ~~1200~~1600px , will need to create responsive designs)
- Changed dark theme background color to #202225 (based on khany's suggestion/design in his forked repo)

## MainContent
- Change appearance of the entire section to a column-by-column flex alongside ControlsContent (2:3 ratio, MainContent:ControlsContent)
- Condensed the rendering of sections by utilizing JS map function
- Changed some styling of the characters so it appears bigger
- Changed input text color to black in dark mode and removed text shadow
- Changed background color of input to use base color plus some opacity (alpha) to make text show more Kedyn/fusliez-notes#24
- Some colors (brown, blue, and purple) are using alternative background colors because the original color lacks contrast to the text color
- Changed the section names. Going with khany's suggestion: suspicious and hit list should fall in the same category. Also changed naming convention of those categories (snake_case to camelCase) #31 

## ControlsContent
- Changed display of switches to a row to minimize the space it was taking up
- Restructured function to reset players, also renamed function and button to reset round instead for clearing purpose
- Added function to reset everything to initial state (can discuss if needed to add a confirm modal or change the color of the button to make it stand out a bit more)
- Changed notes background color to lightgray
- Shrunk notes text area to height 350px

## MapsContent
~~Moved the entire section to be a part of ControlsContent (Seems like most people have a hang of the maps and doesn't require much visual guidance from it)~~
- Changed border color to be green to make the current selection stand out
- Removed draggable characters for now (commented out the code). Really cool feature, but I don't see it being utilized much. There's normally a lot going on in the game and updating the map/characters is not the easiest thing during a round (I apologize in advance if this upsets anyone. again, very easy to revert that code change)
- Added feedback form to MapsContent column (already exists in current published version, just making it official with the push)

### CODE REVIEW

**Comment by francistse23** [comment]
@chilblane @kevinydhan

hey all, made some UI changes to the app and would love to hear what you guys think of it. Kedyn thinks something feels off but can't pinpoint what it is

already talked to Kedyn about limiting 1 issue per PR. I apologize in advance if there's too much to follow. I tried to outline all the changes I have made in the above

Kedyn received feedback that the map almost feels like a cheat so we're thinking of removing them

**Comment by kevinydhan** [comment]
Hey @francistse23, I'm going to take a look at your forked repository and get back to you.

**Comment by chilblane** [comment]
i'm probably going to pass on reviewing this and any other code reviews until i have a chance to jump in and evaluate the project as a whole in the future.

**Comment by francistse23** [comment]
> i'm probably going to pass on reviewing this and any other code reviews until i have a chance to jump in and evaluate the project as a whole in the future.

that's fair. just want to finalize on a design going through it with as many UX/UI devs as possible. a few people did report that they want it to be mobile friendly. I want the majority of the team to agree on a design before I implement responsive design

**Comment by kevinydhan** [comment]
Hey @francistse23, you made some good stylistic and refactoring changes to the application. Unfortunately, I'm confused as to the current status of this project, which is hindering my decision to accept/close this PR. 

@Kedyn, here are the questions I have about the application:
- Which desktop design are we going to implement?
- Are we going to incorporate Redux?
- Are we going to implement a mobile version? If so, can we finalize a design for mobile?

I think it would be best for you to make the executive decision on this before we continue.

**Comment by francistse23** [comment]
To be honest I am a bit lost as well. There's no clear direction where we want to take this app as of right now, except implementing an executable (with Electron) and a mobile version. 

Seems like chilblane and yourself have more UI/UX design experience.  Kedyn and I were waiting on your designs and decide which one to go with. Not sure if you have had a chance to check out [chilblane's mock up](https://www.figma.com/file/2fOS190CofI6dABIRiMLcJ/Among-Us-Notes---Chilblane-Remix?node-id=1%3A170).  

and perhaps Kedyn himself can address the questions you have for him. From my understanding:
- From my conversation with Kedyn, he likes both yours and chilblane's designs.
- We probably won't implement Redux at this moment. It's something to consider but at the current state, maybe useReducer can handle it just fine (if we implement that)
-  I personally want to do a mobile version of this. See below for chilblane's wireframe as an example: 



I'd be down to implement the mobile design, as well as whatever enhancements we'd like to make for the app in general

**Comment by kevinydhan** [comment]
> - We probably won't implement Redux at this moment. It's something to consider but at the current state, maybe useReducer can handle it just fine (if we implement that)

That's fine. I can go ahead and proceed to refactor the `useState` lines into a custom `useReducer` hook.

> - I personally want to do a mobile version of this. See below for chilblane's wireframe as an example:

I'm a fan of that mobile wireframe. I think we can ask @chilblane to flesh out the design to match his desktop version.

As for this PR, it might be best to either cherry pick commits or:
1. Close this PR.
2. Work on the project's restructuring using `useReducer` and better segmented components.
3. Create new scoped tasks on the Issues page. 
4. Assign a developer with a single task and have them submit a PR for that task.

**Comment by Kedyn** [comment]
1. I am a bit concern about the talk of how much of this is a cheat. I feel the score and noting who is who to you, is not much of a cheat since is just what you are thinking during the game. Therefore I think if we make an executable (mostly for those with one screen), it should only have the main content. 

2. I would not mind if someone takes on the task of changing it from context/reducer to redux. I expend quite a bit of time on the redesign of the whole project and it took away from the tempo of the project. I think is fine for now, but open to it if someone takes on the task.

3. From the data collected today from the analytics, we got a few people, around 30%, using mobile... so I think we should create a mobile design.

4. I really like both of your designs, so on the feedback form I have added an image of both designs to see which one people like more. chilblane's is geared more towards general users, yours more towards leslie. I also feel that would not be so crazy to just allow the user to create their own theme.


As for the PR in question... my only concern is that the the layout is more spread... where I think me and both @chilblane and @kevinydhan kept in mind the stream position and size from the viewers prespective.

**Comment by francistse23** [comment]
where'd you hear that this feels like a cheat? I know we had this discussion yesterday about the map. to me, the map itself feels more like a guide, and the notes and players are supplementary to the gameplay like you said

yeah if kevin is up for it, he can certainly change the useState to useReducer to reduce some of the clutter. if not, I can do it using the example he provided

I can do the mobile design unless someone wants to pick it up

I guess we'll wait for some votes on which option users prefer, then go ahead with the design changes?

I am ok with cherry picking commits or scrap my changes, but let's finalize on a design before we make any more changes?

**Comment by Kedyn** [comment]
youtube comments and Cheery also mentioned it

**Comment by francistse23** [comment]
if that's the case and the team is ok with it, I can remove the map from the UI and have the feedback form in a modal.

that should take care of that problem as well as the layout spread you are concerned about

**Comment by Kedyn** [comment]
As previously mentioned, to me: if you have two monitors and you are using both the game and the app's map it has merit because you need a bit more attention to both... but to put it in the electron app should not even be considered. As seen in the feedback people do use the map so I would keep it as is... not adding extra functionality

**Comment by chilblane** [comment]
I would rather see this PR merged in first. If we want to go my design direction, I'd like to get as much engineering work done first. UI can be a second priority.

File: src/components/App.styles.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+export default createUseStyles({
+  container: {
+    display: "flex",
+    maxWidth: "1600px",
+    margin: "0 auto",
+    padding: "0.5rem",
+    width: "100%",
+  },
+});
```
Comment (Kedyn) [inline src/components/App.styles.ts:11]: This could've just gone on "src/themes/default.ts", on the global main section, since is part of the main layout

Comment (francistse23) [reply src/components/App.styles.ts:11]: I'll move that to src/themes/default.ts global

**Pull request merged**